code/model/editableformfields/EditableFormField.php

```diff
@@ -365,6 +365,13 @@ public function canEdit($member = null) {
         $parent = $this->Parent();
 		if($parent && $parent->exists()) {
 			return $parent->canEdit($member) && !$this->isReadonly();
+		} else if ($this->ID == 0) {
+			// This is for GridFieldOrderableRows support as it checks edit permissions on 
+			// singleton of the class. Fixes backwards compatibility issues where 
+			// 'Content Authors' used to be able to re-order items on the old interface 2.x version.
+			$parentClass = $this->config()->has_one['Parent'];
+			$parent = singleton($parentClass);
+			return $parent->canEdit($member);
```

Comment:
I don;t think canEdit should be called on singletons? Can't we do `$this->Parent()->canEdit($member)`